bot/connector-mattermost/src/main/kotlin/MattermostConnector.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2017/2024 e-voyageurs technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.bot.connector.mattermost
+
+import ai.tock.bot.connector.ConnectorBase
+import ai.tock.bot.connector.ConnectorCallback
+import ai.tock.bot.connector.mattermost.model.MattermostMessageIn
+import ai.tock.bot.connector.mattermost.model.MattermostMessageOut
+import ai.tock.bot.engine.BotRepository
+import ai.tock.bot.engine.ConnectorController
+import ai.tock.bot.engine.action.SendSentence
+import ai.tock.bot.engine.event.Event
+import ai.tock.bot.engine.monitoring.logError
+import ai.tock.shared.Executor
+import ai.tock.shared.error
+import ai.tock.shared.injector
+import ai.tock.shared.jackson.mapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.github.salomonbrys.kodein.instance
+import mu.KotlinLogging
+import java.time.Duration
+
+class MattermostConnector(
+    private val applicationId: String,
+    private val path: String,
+    private val url: String,
+    private val token: String,
+    private val channelId: String? = null,
+    private val outgoingToken: String
+) : ConnectorBase(mattermostConnectorType) {
+
+    companion object {
+        private val logger = KotlinLogging.logger {}
+    }
+
+    private val client: MattermostClient = MattermostClient(url)
+    private val executor: Executor by injector.instance()
+
+    override fun register(controller: ConnectorController) {
+        controller.registerServices(path) { router ->
+            logger.info { "Register mattermost service $path" }
+            router.post(path).handler { context ->
+                logger.info { "Handle mattermost request ${context.body().asString()}" }
+                // see https://developers.mattermost.com/integrate/webhooks/outgoing/
+                val requestTimerData = BotRepository.requestTimer.start("mattermost_webhook")
+                try {
+                    val body = when(context.request().getHeader("Content-Type")) {
+                        "application/x-www-form-urlencoded" -> context.convertUrlEncodedStringToJson()
```

Comment:
You should use context.request().formAttributes() ([see doc here](https://vertx.io/docs/apidocs/io/vertx/core/http/HttpServerRequest.html#formAttributes--)) instead of parsing them yourself.

You could simply check if the request as formAttributes.